spec/userLogInSpec.js

```diff
@@ -0,0 +1,26 @@
+"use strict";
+
+describe("user log in tests", () => {
+  let userLogIn;
+  let response;
+
+  it("successfully logs in", () => {
+    let handle = "JasmineTest";
+    let password = "password";
+    response = new Response(
+      JSON.stringify({
+        user_id: 999,
+        session_key: "_2a_12_Pgo1hhpBTcmVE9_oqFHyNu",
+      })
+    );
+    spyOn(window, "fetch").and.returnValue(Promise.resolve(response));
+    userLogIn = new UserLogIn();
+    return userLogIn.logIn(handle, password).then(() => {
+      expect(document.querySelector("#loggedon").innerHTML).toEqual(
```

Comment:
I typically expect tests that deal with async to use the `done()` syntax, but I'm not entirely sure how this works here. It looks good from first glance...
Have you seen it fail, and then pass when you changed you code? Or does it always pass?

Reply:
will check, good point